### PR TITLE
Integrate OpenSSL 3.x

### PR DESCRIPTION
**Pull request by pde-cds**
Created a buildable state including OpenSSL 3.0.12 and Apache Subversion 1.14.3.
Currently only x64,x86 and ARM64 build sucessfully including Unittests (Some Unittest had to be deactivated because they relied on ctf.open.collab.net.)
.Net Core 3.1 builds are also successful but UnitTests fail ~~-> Maybe update to .Net 6.0 or higher?~~.NET 6.0 update seems to be more difficult as expected -> Staying on .NetCore3.1

Necessary steps
- [x] Alternative to svn co for OpenSSL because Github sunsetted the svn support -> using git clone for github based projects
- [x] Either make VSTest for Dotnet Core runnable again or upgrade to Dotnet 6.0 or higher -> ~~Tests for .Net Core 3.1 do run but only on x64 because of https://github.com/actions/setup-dotnet/issues/485~~ Updated SharpSvn.Tests to net6.0 (build artifacts stay on .netcore3.1
- [ ] Check if patches of OpenSSL 1.1.x are also necessary for 3.x because they currently not integrated into the build

### CODE REVIEW

File: .github/workflows/MSBuild.yml [hunk 1]
```diff
@@ -24,9 +25,19 @@ jobs:
     - name: Setup VSTest Path
       uses: darenm/Setup-VSTest@v1.2
 
+    - name: Load dotnet core 3.1.100 and 3.1.x
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: '3.1.x'
+      env:
+        DOTNET_MULTILEVEL_LOOKUP: false
+
```
Comment (rhuijben) [inline .github/workflows/MSBuild.yml:34]: Not sure if we still need this old version. I think 6.0 is the oldest still supported version. We should probable enable 8.0 for the future.

File: .github/workflows/MSBuild.yml [hunk 2]
```diff
@@ -12,7 +13,7 @@ env:
 
 jobs:
   build:
-    runs-on: windows-latest
+    runs-on: windows-2022
```
Comment (rhuijben) [inline .github/workflows/MSBuild.yml:16]: I would assume latest is 2022? Or is that still 2019?

File: src/SharpSvn/SharpSvnStrings.h [hunk 3]
```diff
@@ -1,854 +1,854 @@
...
+    
+    
+    /// <summary>
+    ///   Eine stark typisierte Ressourcenklasse zum Suchen von lokalisierten Zeichenfolgen usw.
+    /// </summary>
+    // Diese Klasse wurde von der StronglyTypedResourceBuilder automatisch generiert
+    // -Klasse über ein Tool wie ResGen oder Visual Studio automatisch generiert.
+    // Um einen Member hinzuzufügen oder zu entfernen, bearbeiten Sie die .ResX-Datei und führen dann ResGen
+    // mit der /str-Option erneut aus, oder Sie erstellen Ihr VS-Projekt neu.
+    [System::CodeDom::Compiler::GeneratedCodeAttribute(L"System.Resources.Tools.StronglyTypedResourceBuilder", L"4.0.0.0"),
```
Comment (rhuijben) [inline src/SharpSvn/SharpSvnStrings.h:22]: Nice... generated the German texts as default ;-)

**Comment by rhuijben** [comment]
Feel free to update to .Net 6.

Talked with Markus. Feel free to get this through... I'll get things released through NuGet when done on master.

**Comment by rhuijben** [comment]
@pde-cds check your mail. Should be invite to project member, so the tests should work directly, etc.

File: .github/workflows/MSBuild.yml (hunk 2 above)
Comment (pde-cds) [reply .github/workflows/MSBuild.yml:16]: Yes windows-latest is currently windows-2022. To ensure there are no implicit changes of the runner "windows-2022" might be the better option

File: src/SharpSvn/SharpSvnStrings.h (hunk 3 above)
Comment (pde-cds) [reply src/SharpSvn/SharpSvnStrings.h]: Reverted changes to the file and commited old version

File: .github/workflows/MSBuild.yml (hunk 1 above)
Comment (pde-cds) [reply .github/workflows/MSBuild.yml]: Updated to net6.0 for now

**Review by rhuijben — APPROVED** [review]
Thanks!

Feel free to merge when done

**Review by shbeck — APPROVED** [review]
Reviewed the changes

**Pull request merged**